### PR TITLE
refactor: interface and markup trip

### PR DESCRIPTION
**Pull request by osminaz**
## Preflight Checklist

- [ x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [ ] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

In order to use this component with our API some changes had to be done

### CODE REVIEW

File: src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx [hunk 1]
```diff
@@ -59,8 +59,8 @@ export class SbbPearlChainTime {
 
     return (
       <div class="sbb-pearl-chain__time">
-        {this.departureWalk && (
-          <span class="sbb-pearl-chain__time-walktime">
+        {this.departureWalk > 0 ? (
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx:62]:
```suggestion
        {this.departureWalk > 0 && (
```

File: src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx [hunk 2]
```diff
@@ -85,8 +87,8 @@ export class SbbPearlChainTime {
           <span class="sbb-screenreaderonly">{i18nArrival[this._currentLanguage]}</span>
           {this.arrivalTime && format(arrival, 'H:mm')}
         </time>
-        {this.arrivalWalk && (
-          <span class="sbb-pearl-chain__time-walktime">
+        {this.arrivalWalk > 0 ? (
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx:90]:
```suggestion
        {this.arrivalWalk > 0 && (
```

File: src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx [hunk 3]
```diff
@@ -96,6 +98,8 @@ export class SbbPearlChainTime {
               <span aria-hidden="true">'</span>
             </time>
           </span>
+        ) : (
+          ''
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx:102]:
```suggestion
        )
```

File: src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx [hunk 4]
```diff
@@ -70,6 +70,8 @@ export class SbbPearlChainTime {
...
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx:74]:
```suggestion
        )
```

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 5]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
-    time: string;
-  };
-  serviceJourney?: {
-    serviceAlteration: ServiceAlteration;
-  };
-};
+
+  /**
+   * Transport-product change from Leg to Leg info according to SBB business rules. Relates to reachable.
+   * example: Your connecting train will be waiting, please change trains immediately.
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:47]:
```suggestion
   * Example: Your connecting train will be waiting, please change trains immediately.
```

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 6]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
+   */
+  reachableText?: string;
+}
+
+/**
+ * A passenger carrying Service (phyisical public transport vehicle)
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:53]:
```suggestion
 * A passenger carrying Service (physical public transport vehicle)
```

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 7]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
+ * provided and operated by a certain Operator allocated to a concrete ServiceJourney
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:54]:
```suggestion
 * provided and operated by a certain operator allocated to a concrete ServiceJourney
```

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 8]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
+ */
+export interface ServiceProduct {
+  /**
+   * Corporate Identity Icon for e.g. Train 'SBB_oev_b_t02'
+   * // TODO - Ist diese Info zum Icon nützlich oder wäre die Art des Transportmittel (ZUg, Bus, ..) nützlicher
+   */
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:60]: Can Todo be avoided? And please in English language.

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 9]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
+   * Usually referring to a specific commercial PT route (where direction might be either way), shown on vehicle displays. If this value is missing, it is probably a single-journey (de:Einzelfahrt)
+   * example: 1
+   */
+  line?: string;
+  /** Product name for e.g. "IC 1 753" */
+  name: string;
+  /**
+   * Unique per OperatingDay and name (where 'IC 1' can run several times per day in either of opposite directions)
+   * example: 753
+   */
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:72]:
```suggestion
  /**
   * Unique per OperatingDay and name (where 'IC 1' can run several times per day in either of opposite directions).
   * Example: 753
   */
```

File: src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts [hunk 10]
```diff
@@ -17,20 +41,104 @@ export type ServiceAlteration = {
...
+  number?: string;
+  /** A characterisation of the public transport operation according to the means of transport. */
+  vehicleMode: VehicleModeEnum;
+  /**
+   * Short, displayable name of product-category (related to Vehicle).
+   * example: IC
```
Comment (jeripeierSBB) [inline src/components/sbb-pearl-chain/sbb-pearl-chain.custom.d.ts:78]:
```suggestion
   * Example: IC
```

File: src/components/sbb-timetable-row/sbb-timetable-row.helper.tsx [hunk 11]
```diff
@@ -35,6 +36,23 @@ export const durationToTime = (duration: number): string => {
   return result.join(' ');
 };
 
+export const convertCauseInIconName = (cause: PtSituationCauseEnum): string => {
```
Comment (jeripeierSBB) [inline src/components/sbb-timetable-row/sbb-timetable-row.helper.tsx:39]: I'm not sure whether this should be done in lyne. But all the pearl chain is very near to your backend API, so maybe it's ok for the moment. @kyubisation ?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx [hunk 12]
```diff
@@ -35,10 +35,10 @@ export class SbbPearlChainTime {
   @Prop() public arrivalTime!: string;
 
   /** Optional prop to render the walk time (in minutes) before departure */
-  @Prop() public departureWalk?: number;
+  @Prop() public departureWalk?: any;
```
[the remark this answers is not in the document]
Comment (osminaz) [reply src/components/sbb-pearl-chain-time/sbb-pearl-chain-time.tsx:38]: Currently the generated types contains in some places the type any, Im sure that we will change it in future. I will add a comment to it for now

**Review by jeripeierSBB — APPROVED** [review]
LGTM

 - Please rebase
 - Please call me to go through chromatic issues (after rebasing)

**Pull request merged**